pkg/controller/labelidentity/label_group_index_test.go

```diff
@@ -0,0 +1,613 @@
+	}
+	i.selectorItems.Update(selectorItemCCopy)
+	i.selectorItems.Update(selectorItemDCopy)
+	for _, tt := range tests {
+		i.DeletePolicySelectors(tt.policyKey)
+		for k, l := range i.labelIdentities {
+			if l.selectorItemKeys.Has(tt.staleSelector) {
+				t.Errorf("Stale selector %s is not deleted from labelMatch %s", tt.staleSelector, k)
+			}
+		}
+		if _, exists, _ := i.selectorItems.GetByKey(tt.staleSelector); exists {
+			t.Errorf("Stale selector %s is not deleted from selectorItem cache", tt.staleSelector)
+		}
+	}
+}
+
+func TestSetPolicySelectors(t *testing.T) {
+	tests := []struct {
+		name             string
+		selectors        []*types.GroupSelector
+		policyKey        string
+		expIDs           []uint32
+		expSelectorItems map[string]selectorItem
+	}{
+		{
+			name: "new selector for policyA",
+			selectors: []*types.GroupSelector{
+				types.NewGroupSelector("", pSelWeb, nil, nil, nil),
+			},
+			policyKey: "policyA",
+			expIDs:    []uint32{1},
+			expSelectorItems: map[string]selectorItem{
+				selectorItemA.selector.NormalizedName: {
+					labelIdentityKeys: sets.NewString(labelA),
+					policyKeys:        sets.NewString("policyA"),
+				},
+			},
+		},
+		{
+			name: "updated selectors for policyA",
+			selectors: []*types.GroupSelector{
+				types.NewGroupSelector("", pSelWeb, nsSelTest, nil, nil),
+				types.NewGroupSelector("", pSelDB, nil, nil, nil),
+			},
+			policyKey: "policyA",
+			expIDs:    []uint32{1, 2, 3},
+			expSelectorItems: map[string]selectorItem{
+				selectorItemB.selector.NormalizedName: {
+					labelIdentityKeys: sets.NewString(labelA),
+					policyKeys:        sets.NewString("policyA"),
+				},
+				selectorItemC.selector.NormalizedName: {
+					labelIdentityKeys: sets.NewString(labelB, labelC),
+					policyKeys:        sets.NewString("policyA"),
+				},
+			},
+		},
+		{
+			name: "existing selector and new selector for policyB",
+			selectors: []*types.GroupSelector{
+				types.NewGroupSelector("", pSelWeb, nil, nil, nil),
+				types.NewGroupSelector("", pSelWeb, nsSelTest, nil, nil),
+			},
+			policyKey: "policyB",
+			expIDs:    []uint32{1},
+			expSelectorItems: map[string]selectorItem{
+				selectorItemA.selector.NormalizedName: {
+					labelIdentityKeys: sets.NewString(labelA),
+					policyKeys:        sets.NewString("policyB"),
+				},
+				selectorItemB.selector.NormalizedName: {
+					labelIdentityKeys: sets.NewString(labelA),
+					policyKeys:        sets.NewString("policyA", "policyB"),
+				},
+				selectorItemC.selector.NormalizedName: {
+					labelIdentityKeys: sets.NewString(labelB, labelC),
+					policyKeys:        sets.NewString("policyA"),
+				},
+			},
+		},
+	}
+	i := NewLabelIdentityIndex()
+	i.labelIdentities = map[string]*labelIdentityMatch{
+		labelA: labelMatchA,
+		labelB: labelMatchB,
+		labelC: labelMatchC,
+	}
+	i.labelIdentityNamespaceIndex = map[string]sets.String{
+		"testing": sets.NewString(labelA, labelB),
+		"nomatch": sets.NewString(labelC),
+	}
```

Comment:
ditto, this should be moved into the loop to avoid side effect being relied, causing concurrent and focused tests failure.

Reply:
Done, and changed test logic as well.